### DISCUSSION TITLE
Cannot access git submodule https://github.com/alefragnani/vscode-bookmarks-core

### DISCUSSION THREAD

**Discussion by shepardo** [alefragnani/vscode-bookmarks#369] [General]
Hi, when I try to update submodules, the submodule vscode-bookmarks-core seems to be private or non existent?

What's proper way to get it?

```
git submodule update --init --recursive
Submodule 'vscode-bookmarks-core' (https://github.com/alefragnani/vscode-bookmarks-core.git) registered for path 'vscode-bookmarks-core'
Submodule 'vscode-whats-new' (https://github.com/alefragnani/vscode-whats-new.git) registered for path 'vscode-whats-new'
Cloning into '/Users/fernando.gonzalez/Documents/src/vscode/vscode-bookmarks/vscode-bookmarks-core'...
remote: Repository not found.
fatal: repository 'https://github.com/alefragnani/vscode-bookmarks-core.git/' not found
fatal: clone of 'https://github.com/alefragnani/vscode-bookmarks-core.git' into submodule path '/Users/fernando.gonzalez/Documents/src/vscode/vscode-bookmarks/vscode-bookmarks-core' failed
Failed to clone 'vscode-bookmarks-core'. Retry scheduled
Cloning into '/Users/fernando.gonzalez/Documents/src/vscode/vscode-bookmarks/vscode-whats-new'...
Cloning into '/Users/fernando.gonzalez/Documents/src/vscode/vscode-bookmarks/vscode-bookmarks-core'...
remote: Repository not found.
fatal: repository 'https://github.com/alefragnani/vscode-bookmarks-core.git/' not found
fatal: clone of 'https://github.com/alefragnani/vscode-bookmarks-core.git' into submodule path '/Users/fernando.gonzalez/Documents/src/vscode/vscode-bookmarks/vscode-bookmarks-core' failed
Failed to clone 'vscode-bookmarks-core' a second time, aborting
```

**Comment by alefragnani** [comment]
Hi @shepardo ,

Right now the core repo is private, and you are unable to access. I started refactoring the extension, to share pieces with my other Numbered Bookmarks extension, but it took more time/effort that I though, so I’m still thinking on the better model to re-open the extension to contributions.

Thank you

**Reply by olejorgenb** [reply]
EDIT: first off - I really like this extension! Thanks for making it :)

BUT - TBH I find this kinda disingenuous (not saying I thinks it's malicious) When I see project on github I expect it to be fully open sourced unless it's part of a obvious open-core like product. vscode extensions can access my code so I prefer to primarily use open sourced extensions.

It's now 4 years later. Any updates or plans to open source the "core"?

Note that it also claims to be "GPL-3.0 © Alessandro Fragnani"  [here](https://marketplace.visualstudio.com/items?itemName=alefragnani.Bookmarks&ssr=false#overview)
